test/screenshot/infra/lib/cbt-api.js

```diff
@@ -400,24 +401,39 @@ https://crossbrowsertesting.com/account
    */
   async killZombieSeleniumTests() {
     // NOTE: This only returns Selenium tests running on the authenticated CBT user's account.
-    // It does NOT return Selenium tests running under other users.
+    // It does NOT return Selenium tests running under other CBT user accounts.
     /** @type {!CbtSeleniumListResponse} */
     const listResponse = await this.sendRequest_(
       getStackTrace('killZombieSeleniumTests'),
       'GET', '/selenium?active=true&num=100'
     );
 
-    const activeSeleniumTestIds = listResponse.selenium.map((test) => test.selenium_session_id);
+    if (!listResponse.selenium) {
+      this.warnMalformedResponse_(listResponse, 'list');
+      return;
+    }
+
+    this.logSeleniumSessions_(listResponse.selenium, 'Active Selenium sessions');
+
+    const activeSeleniumTestIds = listResponse.selenium.map((session) => session.selenium_test_id);
 
     /** @type {!Array<!CbtSeleniumInfoResponse>} */
     const infoResponses = await Promise.all(activeSeleniumTestIds.map((seleniumTestId) => {
       const infoStackTrace = getStackTrace('killZombieSeleniumTests');
       return this.sendRequest_(infoStackTrace, 'GET', `/selenium/${seleniumTestId}`);
     }));
 
-    const stalledSeleniumSessionIds = [];
+    this.logSeleniumSessions_(infoResponses, 'Selenium sessions info');
+
+    const stalledSeleniumTestIds = [];
+
+    for (let i = 0; i < infoResponses.length; i++) {
+      const infoResponse = infoResponses[i];
+      if (!infoResponse.commands) {
+        this.warnMalformedResponse_(infoResponse, 'info', activeSeleniumTestIds[i]);
```

Comment:
What effect does this end up having vs. the script dying on length of undefined before? Does it basically just skip checking one active session?

Reply:
Yes, exactly